schema/collection.js

```diff
@@ -41,13 +42,20 @@ export const CollectionType = new GraphQLObjectType({
         delete gravityOptions.page // this can't also be used with the offset in gravity
         return gravity.with(accessToken, { headers: true })(`collection/${id}/artworks`, gravityOptions)
           .then(({ body, headers }) => {
-            return connectionFromArraySlice(body, options, {
+            const connection = connectionFromArraySlice(body, options, {
               arrayLength: headers["x-total-count"],
               sliceStart: gravityOptions.offset,
             })
+            // Gravity has a bug (see PR https://github.com/artsy/gravity/pull/11240) where the counts
+            // returned in the X-Total-Count header includes unpublished artworks, even though only
+            // published artworks are returned. This can lead to a situation where hasNextPage is true
+            // but endCursor is null, violating Relay norms and causing serious problems for Eigen.
+            connection.pageInfo.hasNextPage =
+              connection.pageInfo.endCursor === null ? false : connection.pageInfo.hasNextPage
```

Comment:
I would add a `FIXME:` that points to the Gravity PR so that this jumps out in our tooling.